### PR TITLE
Merge up from 4 to 5

### PR DESCRIPTION
**Pull request by maxime-rainville**
Just did a merge up from the 4 branch to the 5 branch.

Normally I would just push directly to the branch, but there's enough merge conflicts to resolve that it's probably worth doing an actual peer review.

The following files had conflicts.
```
        both modified:   src/Control/HTTPResponse.php
        deleted by us:   src/View/Parsers/HTML4Value.php
        both modified:   src/View/SSViewer_DataPresenter.php
        both modified:   src/View/Shortcodes/EmbedShortcodeProvider.php
        deleted by us:   tests/php/View/Parsers/HTML4ValueTest.php
```

### CODE REVIEW

File: src/Control/HTTPResponse.php [hunk 1]
```diff
@@ -310,6 +310,14 @@ public function removeHeader($header)
         return $this;
     }
 
+    /**
+     * Sanitise header values to avoid possible XSS vectors
+     */
+    private function sanitiseHeader(string $value): string
```
Comment (maxime-rainville) [inline src/Control/HTTPResponse.php:316]: This method is new from 4 and was conflicting with the strongly typed parameters of redirect in 5

**Review by maxime-rainville — COMMENTED** [review]
Deprecation notice in HTML4Value and HTML4ValueTest were conflicting with the removal of those files in 5.

**Review by maxime-rainville — COMMENTED** [review]
Both [Guy](https://github.com/silverstripe/silverstripe-framework/commit/5b2820e8ace6376a935173d243ed7823504a4eef) and [Chris](https://github.com/silverstripe/silverstripe-framework/commit/31d5aef520f8c8b0c971d87ba62963dd0e1fb9f4) updated `src/View/SSViewer_DataPresenter.php` with very similar fixes.

File: src/View/Shortcodes/EmbedShortcodeProvider.php [hunk 2]
```diff
@@ -16,6 +16,7 @@
 use SilverStripe\View\Parsers\ShortcodeHandler;
 use SilverStripe\View\Parsers\ShortcodeParser;
 use SilverStripe\Control\Director;
+use SilverStripe\Core\Config\Configurable;
```
Comment (maxime-rainville) [inline src/View/Shortcodes/EmbedShortcodeProvider.php:19]: There was a use `use SilverStripe\Dev\Deprecation;` statement here that was causing a conflict.

**Comment by GuySartorelli** [comment]
You mention `src/View/SSViewer_DataPresenter.php` but I don't see that here?
Edit: The incoming changes were rejected, so it doesn't show as a change in the PR.

**Review by GuySartorelli — APPROVED** [review]
I got the same results when doing a local merge up so this looks good to me. For what it's worth, the conflicts were very small, I'd have been comfortable with that being merged and pushed without a PR.

**Pull request merged**